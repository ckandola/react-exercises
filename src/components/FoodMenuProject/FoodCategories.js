import React from 'react';
import PropTypes from 'prop-types';

const FoodCategories = ({categoryList, filterItems}) => {

    return (
        <div className="food-btn-container">
            {categoryList.map(category => {
                return (
                    <button key={category} className="food-filter-btn" onClick={() => filterItems(category)}>{category}</button>
                );
            })}
        </div>
    );
};

export default FoodCategories;

FoodCategories.propTypes = {
    categoryList: PropTypes.arrayOf(PropTypes.string),
    filterItems: PropTypes.func
};
