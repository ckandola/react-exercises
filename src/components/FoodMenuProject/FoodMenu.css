/*
=============== 
Variables
===============
*/

:root {
    --clr-grey-5: hsl(210, 22%, 49%);
    --clr-white: #fff;
    --clr-gold: #c59d5f;
  
    --transition: all 0.3s linear;
    --radius: 0.25rem;
    --max-width: 1170px;
  }
  
  /* section */
  .food-section {
    width: 90vw;
    margin: 0 auto;
    max-width: var(--max-width);
  }
  
  @media screen and (min-width: 992px) {
    .food-section {
      width: 95vw;
    }
  }
  
  /*
  =============== 
  Menu
  ===============
  */
  
  .food-menu {
    padding: 5rem 0;
  }
  .food-title {
    text-align: center;
    margin-bottom: 2rem;
  }
  .food-underline {
    width: 5rem;
    height: 0.25rem;
    background: var(--clr-gold);
    margin-left: auto;
    margin-right: auto;
  }
  .food-btn-container {
    margin-bottom: 4rem;
    display: flex;
    justify-content: center;
  }
  .food-filter-btn {
    background: transparent;
    border-color: transparent;
    font-size: 1rem;
    text-transform: capitalize;
    margin: 0 0.5rem;
    letter-spacing: 1px;
    padding: 0.375rem 0.75rem;
    color: var(--clr-gold);
    cursor: pointer;
    transition: var(--transition);
    border-radius: var(--radius);
  }
  .food-filter-btn:hover {
    background: var(--clr-gold);
    color: var(--clr-white);
  }
  .food-section-center {
    width: 90vw;
    margin: 0 auto;
    max-width: 1170px;
    display: grid;
    gap: 3rem 2rem;
    justify-items: center;
  }
  .food-menu-item {
    display: grid;
    gap: 1rem 2rem;
    max-width: 25rem;
  }
  .food-photo {
    object-fit: cover;
    height: 200px;
    width: 100%;
    border: 0.25rem solid var(--clr-gold);
    border-radius: var(--radius);
    display: block;
  }
  .food-item-info header {
    display: flex;
    justify-content: space-between;
    border-bottom: 0.5px dotted var(--clr-grey-5);
  }
  .food-item-info h4 {
    margin-bottom: 0.5rem;
  }
  .food-price {
    color: var(--clr-gold);
  }
  .food-item-text {
    margin-bottom: 0;
    padding-top: 1rem;
  }
  
  @media screen and (min-width: 768px) {
    .food-menu-item {
      grid-template-columns: 225px 1fr;
      gap: 0 1.25rem;
      max-width: 40rem;
    }
    .food-photo {
      height: 175px;
    }
  }
  @media screen and (min-width: 1200px) {
    .food-section-center {
      width: 95vw;
      grid-template-columns: 1fr 1fr;
    }
    .food-photo {
      height: 150px;
    }
  }
  