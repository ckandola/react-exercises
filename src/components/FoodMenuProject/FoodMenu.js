import React from 'react';
import PropTypes from 'prop-types';

const FoodMenu = ({items}) => {
    return (
        <div className="food-section-center">
            {items.map(item => {
                const {id, title, img, desc, price} = item;
                return <article key={id} className="food-menu-item">
                    <img src={img} alt={title} className="food-photo" />
                    <div className="food-item-info">
                        <header>
                            <h4>{title}</h4>
                            <h4 className="food-price">${price}</h4>
                        </header>
                        <p className="food-item-text">{desc}</p>
                    </div>
                </article>
            })}
        </div>
    );
};

export default FoodMenu;

FoodMenu.propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape)
};
