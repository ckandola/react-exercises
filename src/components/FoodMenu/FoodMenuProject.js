import React, {useState} from 'react';
import './index.css';
import FoodMenu from './index';
import FoodCategories from './FoodCategories';
import foodItems from './data';

const FoodMenuProject = () => {
    const [menuItems, setMenuItems] = useState(foodItems);
    const categoryList = ['all', ...new Set(foodItems.map(item => item.category))];

    const filterItems = category => {
        if (category === 'all') {
            setMenuItems(foodItems);
            return;
        }
        const newItems = foodItems.filter(item => item.category === category);
        setMenuItems(newItems);
    };

    return (
        <main>
          <section className="food-menu food-section">
            <div className="food-title">
              <h2>Our Menu</h2>
              <div className="underline"></div>
            </div>
            <FoodCategories filterItems={filterItems} categoryList={categoryList} />
            <FoodMenu items={menuItems}/>
          </section>
        </main>
    );
}

export default FoodMenuProject;
