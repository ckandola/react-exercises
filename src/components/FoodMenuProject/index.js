// from https://youtu.be/a_7Z7C_JCyo?t=5299

import React, {useState} from 'react';
import './FoodMenu.css';
import FoodMenu from './FoodMenu';
import FoodCategories from './FoodCategories';
import foodItems from './data';

const FoodMenuProject = () => {
    const [menuItems, setMenuItems] = useState(foodItems);
    const categoryList = ['all', ...new Set(foodItems.map(item => item.category))];

    const filterItems = category => {
        if (category === 'all') {
            setMenuItems(foodItems);
            return;
        }
        const newItems = foodItems.filter(item => item.category === category);
        setMenuItems(newItems);
    };

    return (
        <main>
          <section className="food-menu food-section">
            <div className="food-title">
              <h2>Our Menu</h2>
              <div className="food-underline"></div>
            </div>
            <FoodCategories filterItems={filterItems} categoryList={categoryList} />
            <FoodMenu items={menuItems}/>
          </section>
        </main>
    );
}

export default FoodMenuProject;
