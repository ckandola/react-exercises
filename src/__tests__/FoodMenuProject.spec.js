import React from 'react';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import FoodMenuProject from '../components/FoodMenuProject';

configure({adapter: new Adapter()});

const totalItems = 9;
const totalShakes = 3;
const wrapper = mount(<FoodMenuProject />);

describe('FoodMenuProject tests', () => {
    it('matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it(`displays ${totalItems} items overall, but only ${totalShakes} shakes.`, () => {
        expect(wrapper.find('.food-menu-item')).toHaveLength(totalItems);
        wrapper.find({children: 'shakes'}).simulate('click');
        expect(wrapper.find('.food-menu-item')).toHaveLength(totalShakes);
    });

    it(`returns to all ${totalItems} items after \`all\` is pressed again.`, () => {
        expect(wrapper.find('.food-menu-item')).toHaveLength(totalShakes);
        wrapper.find({children: 'all'}).simulate('click');
        expect(wrapper.find('.food-menu-item')).toHaveLength(totalItems);
    });
});
