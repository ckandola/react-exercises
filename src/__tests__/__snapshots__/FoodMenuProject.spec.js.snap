// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodMenuProject tests matches the snapshot 1`] = `
<FoodMenuProject>
  <main>
    <section
      className="food-menu food-section"
    >
      <div
        className="food-title"
      >
        <h2>
          Our Menu
        </h2>
        <div
          className="food-underline"
        />
      </div>
      <FoodCategories
        categoryList={
          Array [
            "all",
            "breakfast",
            "lunch",
            "shakes",
          ]
        }
        filterItems={[Function]}
      >
        <div
          className="food-btn-container"
        >
          <button
            className="food-filter-btn"
            key="all"
            onClick={[Function]}
          >
            all
          </button>
          <button
            className="food-filter-btn"
            key="breakfast"
            onClick={[Function]}
          >
            breakfast
          </button>
          <button
            className="food-filter-btn"
            key="lunch"
            onClick={[Function]}
          >
            lunch
          </button>
          <button
            className="food-filter-btn"
            key="shakes"
            onClick={[Function]}
          >
            shakes
          </button>
        </div>
      </FoodCategories>
      <FoodMenu
        items={
          Array [
            Object {
              "category": "breakfast",
              "desc": "I'm baby woke mlkshk wolf bitters live-edge blue bottle, hammock freegan copper mug whatever cold-pressed ",
              "id": 1,
              "img": "./images/item-1.jpeg",
              "price": 15.99,
              "title": "buttermilk pancakes",
            },
            Object {
              "category": "lunch",
              "desc": "vaporware iPhone mumblecore selvage raw denim slow-carb leggings gochujang helvetica man braid jianbing. Marfa thundercats ",
              "id": 2,
              "img": "./images/item-2.jpeg",
              "price": 13.99,
              "title": "diner double",
            },
            Object {
              "category": "shakes",
              "desc": "ombucha chillwave fanny pack 3 wolf moon street art photo booth before they sold out organic viral.",
              "id": 3,
              "img": "./images/item-3.jpeg",
              "price": 6.99,
              "title": "godzilla milkshake",
            },
            Object {
              "category": "breakfast",
              "desc": "Shabby chic keffiyeh neutra snackwave pork belly shoreditch. Prism austin mlkshk truffaut, ",
              "id": 4,
              "img": "./images/item-4.jpeg",
              "price": 20.99,
              "title": "country delight",
            },
            Object {
              "category": "lunch",
              "desc": "franzen vegan pabst bicycle rights kickstarter pinterest meditation farm-to-table 90's pop-up ",
              "id": 5,
              "img": "./images/item-5.jpeg",
              "price": 22.99,
              "title": "egg attack",
            },
            Object {
              "category": "shakes",
              "desc": "Portland chicharrones ethical edison bulb, palo santo craft beer chia heirloom iPhone everyday",
              "id": 6,
              "img": "./images/item-6.jpeg",
              "price": 18.99,
              "title": "oreo dream",
            },
            Object {
              "category": "breakfast",
              "desc": "carry jianbing normcore freegan. Viral single-origin coffee live-edge, pork belly cloud bread iceland put a bird ",
              "id": 7,
              "img": "./images/item-7.jpeg",
              "price": 8.99,
              "title": "bacon overflow",
            },
            Object {
              "category": "lunch",
              "desc": "on it tumblr kickstarter thundercats migas everyday carry squid palo santo leggings. Food truck truffaut  ",
              "id": 8,
              "img": "./images/item-8.jpeg",
              "price": 12.99,
              "title": "american classic",
            },
            Object {
              "category": "shakes",
              "desc": "skateboard fam synth authentic semiotics. Live-edge lyft af, edison bulb yuccie crucifix microdosing.",
              "id": 9,
              "img": "./images/item-9.jpeg",
              "price": 16.99,
              "title": "quarantine buddy",
            },
          ]
        }
      >
        <div
          className="food-section-center"
        >
          <article
            className="food-menu-item"
            key="1"
          >
            <img
              alt="buttermilk pancakes"
              className="food-photo"
              src="./images/item-1.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  buttermilk pancakes
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  15.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                I'm baby woke mlkshk wolf bitters live-edge blue bottle, hammock freegan copper mug whatever cold-pressed 
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="2"
          >
            <img
              alt="diner double"
              className="food-photo"
              src="./images/item-2.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  diner double
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  13.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                vaporware iPhone mumblecore selvage raw denim slow-carb leggings gochujang helvetica man braid jianbing. Marfa thundercats 
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="3"
          >
            <img
              alt="godzilla milkshake"
              className="food-photo"
              src="./images/item-3.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  godzilla milkshake
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  6.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                ombucha chillwave fanny pack 3 wolf moon street art photo booth before they sold out organic viral.
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="4"
          >
            <img
              alt="country delight"
              className="food-photo"
              src="./images/item-4.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  country delight
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  20.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                Shabby chic keffiyeh neutra snackwave pork belly shoreditch. Prism austin mlkshk truffaut, 
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="5"
          >
            <img
              alt="egg attack"
              className="food-photo"
              src="./images/item-5.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  egg attack
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  22.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                franzen vegan pabst bicycle rights kickstarter pinterest meditation farm-to-table 90's pop-up 
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="6"
          >
            <img
              alt="oreo dream"
              className="food-photo"
              src="./images/item-6.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  oreo dream
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  18.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                Portland chicharrones ethical edison bulb, palo santo craft beer chia heirloom iPhone everyday
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="7"
          >
            <img
              alt="bacon overflow"
              className="food-photo"
              src="./images/item-7.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  bacon overflow
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  8.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                carry jianbing normcore freegan. Viral single-origin coffee live-edge, pork belly cloud bread iceland put a bird 
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="8"
          >
            <img
              alt="american classic"
              className="food-photo"
              src="./images/item-8.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  american classic
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  12.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                on it tumblr kickstarter thundercats migas everyday carry squid palo santo leggings. Food truck truffaut  
              </p>
            </div>
          </article>
          <article
            className="food-menu-item"
            key="9"
          >
            <img
              alt="quarantine buddy"
              className="food-photo"
              src="./images/item-9.jpeg"
            />
            <div
              className="food-item-info"
            >
              <header>
                <h4>
                  quarantine buddy
                </h4>
                <h4
                  className="food-price"
                >
                  $
                  16.99
                </h4>
              </header>
              <p
                className="food-item-text"
              >
                skateboard fam synth authentic semiotics. Live-edge lyft af, edison bulb yuccie crucifix microdosing.
              </p>
            </div>
          </article>
        </div>
      </FoodMenu>
    </section>
  </main>
</FoodMenuProject>
`;
